eel/src/invoice.rs

```diff
@@ -21,6 +21,12 @@ use log::info;
 use perro::{invalid_input, MapToError, MapToErrorForUnitType, ResultTrait};
 use secp256k1::ecdsa::RecoverableSignature;
 
+#[derive(PartialEq, Eq, Debug)]
+pub(crate) enum InvoiceType {
```

Comment:
What if `Invoice.amount_milli_satoshis()` returns `Some(0)`? You think the enum makes the code less idiomatic rather than more?

Changing this would mean finding a better way for [this check](https://github.com/getlipa/lipa-lightning-lib/blob/c94fdb1584fcba004be43c279558cb59e3976385/eel/src/lib.rs#L520) as well. Maybe splitting the `validate_persist_new_outgoing_payment_attempt` fn into two as well,  one for _regular invoices_ and one for _open invoices_.

I tend to agree with your point, just wanted to make sure you are aware of these circumstances.

Reply:
Let me try to improve this.